mouseless: init at 0.2.0

imsuck opened the pull request:
A replacement for the mouse in Linux.

Repo: https://github.com/jbensmann/mouseless

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,32 @@
+  src = fetchFromGitHub {
+    owner = "jbensmann";
+    repo = "mouseless";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

Aleksanaa commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,32 @@
+    hash = "sha256-iDSTV2ugvHoBuQWmMg2ILXP/Mlt7eq5B2dVaB0jwJOE=";
+  };
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

imsuck replied:
Should it be `meta.license = lib.licenses.mit`?

Aleksanaa commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,32 @@
+  };
+
+  meta = with lib; {
+    description = "A replacement for the mouse in Linux";
```
```suggestion
    description = "Replacement for the mouse in Linux";
```

imsuck replied:
This I just copied from the repo's About section. I think I'll keep it like that. 😅

awwpotato replied:
the [nixpkgs manual explicitly states](https://nixos.org/manual/nixpkgs/unstable/#var-meta-description) not to start your description with "A"

imsuck replied:
Ah thank you.

Aleksanaa commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,32 @@
+
+  meta = with lib; {
+    description = "A replacement for the mouse in Linux";
+    longDescription = ''
+      This program allows you to control the mouse pointer in Linux
+      with the keyboard. It works in all Linux distributions, even those
+      running with Wayland.'';
```
```suggestion
    longDescription = ''
      This program allows you to control the mouse pointer in Linux
      with the keyboard. It works in all Linux distributions, even those
      running with Wayland.
    '';
```

Aleksanaa commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,32 @@
+      with the keyboard. It works in all Linux distributions, even those
+      running with Wayland.'';
+    homepage = "https://github.com/jbensmann/mouseless";
+    license = licenses.mit;
```
New packages must have `maintainers`. Package containing only 1 executable should also have `mainProgram`.

awwpotato commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,35 @@
+      with the keyboard. It works in all Linux distributions, even those
+      running with Wayland.
+    '';
+    homepage = "https://github.com/jbensmann/mouseless";
```
```suggestion
    homepage = "https://github.com/jbensmann/mouseless";
    changelog = "https://github.com/jbensmann/mouseless/releases/tag/v${version}";
```

awwpotato commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,36 @@
+    homepage = "https://github.com/jbensmann/mouseless";
+    changelog = "https://github.com/jbensmann/mouseless/releases/tag/v${version}";
+    maintainers = with lib.maintainers; [ imsuck ];
+    license = lib.licenses.mit;
```
```suggestion
    license = lib.licenses.mit;
    platforms = lib.platforms.linux;
```

awwpotato commented on pkgs/by-name/mo/mouseless/package.nix:
```diff
@@ -0,0 +1,37 @@
+  lib,
+}:
+
+let
+  version = "0.2.0";
+in
+buildGoModule {
+  pname = "mouseless";
+  inherit version;
+
+  vendorHash = "sha256-2q7L9BVcAaT4h/vUcNjVc5nOAFnb4J3WabcEGxI+hsA=";
+
+  src = fetchFromGitHub {
+    owner = "jbensmann";
+    repo = "mouseless";
+    tag = "v${version}";
```
Generally the preferred way to do this is with finalAttrs (see #315337), but you welcome to keep it like this if that's what you prefer.
```suggestion
buildGoModule (finalAttrs: {
  pname = "mouseless";
  version = "0.2.0";
  
  vendorHash = "sha256-2q7L9BVcAaT4h/vUcNjVc5nOAFnb4J3WabcEGxI+hsA=";

  src = fetchFromGitHub {
    owner = "jbensmann";
    repo = "mouseless";
    tag = "v${finalAttrs.version}";
```

0xda157 reviewed (commented):
All of the `mouseless` commits should be squashed into `mouseless: init at 0.2.0`. Aside from that everything looks good.

awwpotato commented:
Adding yourself to the maintainer list should be a separate commit (ideally before the commit adding mouseless)

awwpotato commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 393066`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mouseless</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>mouseless</li>
  </ul>
</details>

imsuck committed:
maintainers: add imsuck

imsuck committed:
mouseless: init at 0.2.0

mouseless: update metadata

mouseless: add changelog to metadata
Co-authored-by: awwpotato <153149335+awwpotato@users.noreply.github.com>

mouseless: follow convention for description

mouseless: add `platforms = linux` (metadata)
Co-authored-by: awwpotato <153149335+awwpotato@users.noreply.github.com>

mouseless: use `finalAttrs` instead of `let in`
